src/librustdoc/clean/types.rs

```diff
@@ -406,7 +406,7 @@ impl Item {
             // were never supposed to work at all.
             let stab = self.stability(tcx)?;
             if let rustc_attr_parsing::StabilityLevel::Stable {
-                allowed_through_unstable_modules: true,
+                allowed_through_unstable_modules: Some(_),
                 ..
             } = stab.level
             {
```

Comment:
@notriddle I am not sure what you mean by "translate" in your comment below. rustc itself also does not currently consider all of these paths deprecated, but it seems like a good idea to have rustdoc de-emphasize all of them as a form of soft deprecation.

Reply:
I meant to set `note: Some(deprecation_message)`, because, if `note` were set, it would show up here.

![image](https://github.com/user-attachments/assets/19f1f42c-e443-47bb-bdf4-63923b75f956)

Instead of just saying <kbd>:-1: Deprecated</kbd>, I'd like to show the same message that's shown in the compiler, since that points you at `mem::transmute`.

Reply:
Not all of these will have a message though, so things will still look odd for those that don't. It seems nice to have rustdoc de-emphasize these items even before we have a proper deprecation message for them, but I am not sure how to do that. I can easily change the code here to only show "deprecated" when there is a message, but then the "wrong" path would show up in search results again. Would be nice to have a sort of soft deprecation where rustc doesn't warn but also rustdoc doesn't point people there.

Reply:
Can you give an example where (1) no deprecation message is being added (2) the "bad" path is shorter than, or the same length as, the stable one? If there aren't any, then nothing needs to be done and we can just remove the deprecation tag whenever the message is None, because intra-doc links and search results both already prefer shorter paths.

Reply:
> intra-doc links and search results both already prefer shorter paths.

That doesn't seem to be true, given that `intrinsics::transmute` is preferred over `mem::transmute` (before your patch).

Reply:
It's counting components, not characters.

Reply:
Ah. I haven't yet checked the other uses of allowed_through_unstable_modules; I plan to do that once this PR lands.